### PR TITLE
Upgrade to PHP 5.6 with Zend OpCache and APCu

### PR DESCRIPTION
**Pull request by renatomassaro**
- Upgrade to PHP 5.6
- Add "new" Zend OpCache
- Add "new" APCu cache (faster than APC)
- Install ImageMagick bin (used by Phabricator for meme creation)

Rebuilt from scratch and all is working well

### CODE REVIEW

**Comment by hach-que** [comment]
As per the comment in #13, this image needs to be migrated towards a derive and bake model before I can merge in more major changes.

**Comment by lamjack** [comment]
can i search this image in docker repository?

File: server-http.conf [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+        access_log /var/log/nginx/access.log combined buffer=256k flush=5m;
+        error_log /var/log/nginx/error.log;
 
         root /srv/phabricator/phabricator/webroot;
         try_files $uri $uri/ /index.php;
 
-        location /.well-known/ {
-            root /srv/letsencrypt-webroot;
-        }
+        #location /.well-known/ {
```
Comment (hach-que) [inline server-http.conf]: We can't remove this as it's required for SSL to work.

File: Dockerfile [hunk 2]
```diff
@@ -60,26 +76,6 @@ RUN git clone --progress git://github.com/PHPOffice/PHPExcel.git
 USER root
 WORKDIR /
 
-# Clone Let's Encrypt
-RUN git clone https://github.com/letsencrypt/letsencrypt /srv/letsencrypt
-
-# Install WinRM tool
-RUN zypper --non-interactive in go
```
Comment (hach-que) [inline Dockerfile]: This is needed for Windows Drydock hosts to work.

File: Dockerfile [hunk 3]
```diff
@@ -60,26 +76,6 @@ RUN git clone --progress git://github.com/PHPOffice/PHPExcel.git
 USER root
 WORKDIR /
 
-# Clone Let's Encrypt
-RUN git clone https://github.com/letsencrypt/letsencrypt /srv/letsencrypt
```
Comment (hach-que) [inline Dockerfile]: This is needed for SSL to work.

File: 20-postfix [hunk 4]
```diff
@@ -1,6 +0,0 @@
-#!/bin/bash
-
-# Run Postfix
-export SYSTEMD_NO_WRAP=true
-/etc/init.d/postfix start
```
Comment (hach-que) [inline 20-postfix]: This is needed for emails to be sent.

**Comment by hach-que** [comment]
There are some issues with this change, and it needs to be rebased on the latest version in order to be merged.

**Comment by renatomassaro** [comment]
Yep, this PR changes were supposed to go only until commit  	f02e2d1. Not sure how I change the PR HEAD here on Github.

Since you said this image must be rewritten, I decided to reshape my fork to my own needs.

**Comment by hach-que** [comment]
Just an FYI, I'm working on bringing these changes upstream, along with a number of configuration improvements.  In particular:

* The base image is upgraded to OpenSUSE Leap 42.1
* It'll have PHP 5.6 and APCu installed
* The number of layers in the Docker image are being significantly reduced
* You'll be able to configure it the conventional way with /config
* You'll be able to configure it by setting environment variables
* You'll be able to configure it by deriving your own image from this image, and running setup commands to bake the configuration in

I'll be updating the README with the new configuration options when this work is done.

**Comment by hach-que** [comment]
@renatomassaro All of these changes are now available in this image, and it now supports configuration via environment variables as well as scripts.

Going to close this PR as there's nothing left to do here.

**Pull request closed without merging**